### DISCUSSION TITLE
DuckDB silently stopped without errors but not complete the task

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#5527] [General]
in Windows 10, using this script, DuckDB simply stop working after less than a minute, the files are not written back to disk, using (0.6.1.dev160), the problem happen only in Windows 10, tried the same code in Colab and it works fine. 

```
import pyarrow.dataset as ds
import duckdb 
con = duckdb.connect()
result =con.execute(
'''
SET memory_limit='10GB';
select *,extract ( year from l_shipdate) as year  from 'C:/TPC-H-SF10/Parquet/10/lineitem.parquet' order by l_shipdate
''').fetch_record_batch()
ds.write_dataset(result,"C:/lineitem", format="parquet" , partitioning=['year'],partitioning_flavor="hive",max_rows_per_group=120000,existing_data_behavior="overwrite_or_ignore")
```

the source file is here
  `!curl -L 'https://drive.google.com/uc?export=download&id=18gv0Yd_a-Zc7CSolol8qeYVAAzSthnSN&confirm=t' > lineitem.parquet`

**Comment by hannes** [comment]
What do you mean "stop working"? Is it crashing?

**Reply by djouallah** [reply]
I guess it does crash, but it is not showing any error

**Reply by Mause** [reply]
> I guess it does crash, but it is not showing any error

Does it max out the cpu or memory in task manager first? Does it show a "this program has stopped working" window? Do you happen to have a crash dump?

**Reply by djouallah** [reply]
no, it just show that the script has finished, but the data is not written back to the disk, the script should take 7 minutes in my laptop but it does finish in less than a minutes, the code I provided is reproducible

**Reply by Mause** [reply]
> no, it just show that the script has finished, but the data is not written back to the disk, the script should take 7 minutes in my laptop but it does finish in less than a minutes, the code I provided is reproducible

Can I ask how much RAM the machine you're testing with has? I've been trying your test case on Github actions machines but running into allocation errors

**Reply by djouallah** [reply]
16 GB

**Comment by djouallah** [comment]
I used faulthandler.enable() and now it show the error

`duckdb.IOException: IO Error: Could not read file ".tmp\duckdb_temp_storage-0.tmp" (error in ReadFile(location: 284950528, nr_bytes: 262144)): Reached the end of the file.`

**Reply by djouallah** [reply]
using 0.6.1.dev245

```
Windows fatal exception: access violation

Thread 0x00005eac (most recent call first):
  File "c:\Users\mimoune.djouallah\Desktop\TPC-H-SF10\0-Code\export_partition.py", line 6 in <module>
Windows fatal exception: access violation

Current thread 0x00005eac (most recent call first):
  File "c:\Users\mimoune.djouallah\Desktop\TPC-H-SF10\0-Code\export_partition.py", line 6 in <module>
Fatal Python error: Segmentation fault

Current thread 0x00005eac (most recent call first):
  File "c:\Users\mimoune.djouallah\Desktop\TPC-H-SF10\0-Code\export_partition.py", line 6 in <module>

Extension modules: numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, pyarrow.lib, pyarrow._hdfsio, pyarrow._fs, pyarrow._compute, 
pyarrow._csv, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.strptime, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.tslib, pandas._libs.lib, pandas._libs.hashing, pandas._libs.ops, pandas._libs.arrays, pandas._libs.index, pandas._libs.join, pandas._libs.sparse, pandas._libs.reduction, pandas._libs.indexing, pandas._libs.internals, pandas._libs.writers, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.testing, pandas._libs.parsers, pandas._libs.json, pyarrow._dataset, pyarrow._parquet, pyarrow._dataset_parquet (total: 62)
Windows fatal exception: access violation

Current thread 0x00005eac (most recent call first):
```

**Reply by Mause** [reply]
> using 0.6.1.dev245
> 
> ```
> Windows fatal exception: access violation
> 
> Thread 0x00005eac (most recent call first):
>   File "c:\Users\mimoune.djouallah\Desktop\TPC-H-SF10\0-Code\export_partition.py", line 6 in <module>
> Windows fatal exception: access violation
> 
> Current thread 0x00005eac (most recent call first):
>   File "c:\Users\mimoune.djouallah\Desktop\TPC-H-SF10\0-Code\export_partition.py", line 6 in <module>
> Fatal Python error: Segmentation fault
> 
> Current thread 0x00005eac (most recent call first):
>   File "c:\Users\mimoune.djouallah\Desktop\TPC-H-SF10\0-Code\export_partition.py", line 6 in <module>
> 
> Extension modules: numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, pyarrow.lib, pyarrow._hdfsio, pyarrow._fs, pyarrow._compute, 
> pyarrow._csv, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.strptime, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.tslib, pandas._libs.lib, pandas._libs.hashing, pandas._libs.ops, pandas._libs.arrays, pandas._libs.index, pandas._libs.join, pandas._libs.sparse, pandas._libs.reduction, pandas._libs.indexing, pandas._libs.internals, pandas._libs.writers, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.testing, pandas._libs.parsers, pandas._libs.json, pyarrow._dataset, pyarrow._parquet, pyarrow._dataset_parquet (total: 62)
> Windows fatal exception: access violation
> 
> Current thread 0x00005eac (most recent call first):
> ```

You ran the same code and got this failure instead?

**Reply by djouallah** [reply]
yes

**Comment by hannes** [comment]
Trying to reproduce this on AWS today

**Reply by djouallah** [reply]
No, the issue is only with Windows, it works fine in Linux

**Reply by hannes** [reply]
They do have Windows boxes on AWS ^^

**Comment by hannes** [comment]
I'm getting 
`
duckdb.IOException: IO Error: Could not read file ".tmp\duckdb_temp_storage-0.tmp" (error in ReadFile(location: 40108032, nr_bytes: 262144)): Reached the end of the file.
` Indeed looking like a bug.

**Comment by hannes** [comment]
We should have tracked this down https://github.com/duckdb/duckdb/pull/5600

**Reply by djouallah** [reply]
@hannes thank you, it works now
